xbmc/addons/AddonManager.cpp

```diff
@@ -242,7 +242,12 @@ VECADDONS CAddonMgr::GetAvailableUpdates() const
   {
     const auto& remote = last_versions.find(addon->ID());
     if (remote != last_versions.end() && remote->second->Version() > addon->Version())
-      updates.emplace_back(std::move(remote->second));
+    {
+      if (IsUpdateFromTrustedRepo(addon, remote->second->Origin()))
+      {
+        updates.emplace_back(std::move(remote->second));
+      }
+    }
```

Comment:
I think you have two problems here. 

1) Lets say you have an addon from a trusted repo. Then a private repo is installed that has higher version of this addon. It won't install the the addon from the private repo which is correct, but it won't installed any future updates from the trusted repo either.

2) What about updates from non trusted repos?

Reply:
both points clear now